lib/msf/core/post/file.rb

```diff
@@ -692,4 +683,43 @@ def _unix_max_line_length
 
     line_max
   end
+
+
+  def stat(filename)
+     if session.type == 'meterpreter'
+      return session.fs.file.stat(filename)
+    end
+    if session.type == 'shell'
+      if session.platform == 'windows'
+        raise "`stat' method does not support Windows systems"
+      end
+      return FileStat.new(filename, session)
+    end
+  end
+
+
+  class FileStat < Rex::Post::FileStat
+  
+    attr_accessor :stathash
+
+    def initialize(filename, session)
+      data = session.shell_command_token("stat --format='%d,%i,%h,%u,%g,%t,%s,%B,%b,%x,%y,%z,%f' #{filename}").to_s.chomp
+      data = data.split(",")
```

Comment:
This should use a regex or something to validate the results returned by the command are expected. My concern is that the format argument may not behave the same way on every system.

Reply:
Okay, I'll add a check for that.

Reply:
This should raise an exception so the instance isn't left uninitialized.